tests/native_api_tests/test_api_model.cpp

```diff
@@ -494,14 +494,23 @@ TEST_CASE("API Model") {
         update_sym_load_buffer.set_value(PGM_def_update_sym_load_id, update_sym_load_id.data(), -1);
         update_sym_load_buffer.set_value(PGM_def_update_sym_load_q_specified, update_sym_load_q_specified.data(), -1);
 
+        // update dataset buffers - no ids
+        Buffer update_source_buffer_no_id{PGM_def_update_source, 2};
+        update_source_buffer_no_id.set_nan();
+        update_source_buffer_no_id.set_value(PGM_def_update_source_u_ref, update_source_u_ref.data(), -1);
+
+        Buffer update_sym_load_buffer_no_id{PGM_def_update_sym_load, 2};
+        update_sym_load_buffer_no_id.set_nan();
```

Comment:
Nitpick: I know it's good practice, but should we also test without setting first to `nan`? It should also work as intended, right?